test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	"antrea.io/antrea/pkg/features"
+	. "antrea.io/antrea/test/e2e/utils"
+)
+
+const interfaceNameLength = 5
+
+type vmInfo struct {
+	nodeName string
+	osType   string
+	ifName   string
+	ip       string
+	ifIndex  string // Used only for Windows
+}
+
+// TestVMAgent is the top-level test which can contain some subtests for
+// VMAgent so they can share setup, teardown.
+func TestVMAgent(t *testing.T) {
+	skipIfFeatureDisabled(t, features.ExternalNode, false, true)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	t.Run("testExternalNodeBasic", func(t *testing.T) { testExternalNodeBasic(t, data, "vm-ns") })
+}
+
+func testExternalNodeBasic(t *testing.T, data *TestData, namespace string) {
+	t.Logf("List of Windows VMs: '%s', Linux VMs: '%s'", testOptions.winVMs, testOptions.linVMs)
+	nsReturned, _ := data.clientset.CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{})
+	saReturned, _ := data.clientset.CoreV1().ServiceAccounts(namespace).Get(context.TODO(), "vm-agent", metav1.GetOptions{})
+	t.Logf("Using service account %s", saReturned.Name)
+	t.Logf("Using namespace %s", nsReturned.Name)
+	var vmList []vmInfo
+	if testOptions.linVMs != "" {
+		vms := strings.Split(testOptions.linVMs, ",")
+		for _, vm := range vms {
+			t.Logf("Get info for Linux VM: %s", vm)
+			tempVm := getVMInfo(t, data, vm)
+			vmList = append(vmList, tempVm)
+		}
+	}
+	if testOptions.winVMs != "" {
+		vms := strings.Split(testOptions.winVMs, ",")
+		for _, vm := range vms {
+			t.Logf("Get info for Windows VM, %s", vm)
+			tempVm := getWindowsVMInfo(t, data, vm)
+			vmList = append(vmList, tempVm)
+		}
+	}
+
+	t.Logf("TestSetup")
+	for _, vm := range vmList {
+		stopAntreaAgent(t, data, vm)
+		t.Logf("Creating ExternalNode for %+v", vm)
+		_, err := createExternalNodeCRD(data, nsReturned.Name, vm.nodeName, vm.ifName, vm.ip)
```

Comment:
Use `namespace` directly?

Reply:
Sure, for now will add a global variable for nameSpace and serviceAccount